Skip annotate boolean input

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #2957

Differential Revision: [D55946526](https://our.internmc.facebook.com/intern/diff/D55946526/)

cccclai committed:
Skip annotate boolean input



Differential Revision: [D55946526](https://our.internmc.facebook.com/intern/diff/D55946526/)

[ghstack-poisoned]

chiwwang commented:
@shewu-quic @haowhsu-quic

cccclai committed:
Update on "Skip annotate boolean input"


Differential Revision: [D55946526](https://our.internmc.facebook.com/intern/diff/D55946526/)

[ghstack-poisoned]

shewu-quic commented:
Look good for me. Thanks for your help.

jerryzh168 commented on backends/qualcomm/quantizer/utils.py:
```diff
@@ -41,6 +42,14 @@ def decorator(annotator: Callable):
+    since observers only works with float Tensors
+    """
+    if "val" not in node.meta or not isinstance(node.meta["val"], FakeTensor):
+        return True
```
why is this True?

jerryzh168 commented on backends/qualcomm/quantizer/utils.py:
```diff
@@ -123,11 +132,11 @@ def annotate_binary(node: Node, quantization_config: QuantizationConfig) -> None
     input_qspec_map = {}
     input_act0 = node.args[0]
-    if isinstance(input_act0, Node):
+    if isinstance(input_act0, Node) and _is_input_float_tensor(input_act0):
```
nit: we could also move `isinstance(input_act0, Node)` check into `_is_input_float_tensor` function as well

cccclai committed:
Update on "Skip annotate boolean input"


Differential Revision: [D55946526](https://our.internmc.facebook.com/intern/diff/D55946526/)

[ghstack-poisoned]

cccclai committed:
Update on "Skip annotate boolean input"


Differential Revision: [D55946526](https://our.internmc.facebook.com/intern/diff/D55946526/)

[ghstack-poisoned]

cccclai commented:
@pytorchbot cherry-pick --onto release/0.2 -c critical
